### PR TITLE
build: automate CloudFormation deployment and upgrade to Node 20

### PR DESCRIPTION
**Pull request by AndreaDiotallevi**
## What does this change?

[Trello](https://trello.com/c/VFDNSPhF/903-upgrade-fulfilment-lambdas-to-use-node-20-and-automate-deployment)

- Automate the CloudFormation deployment
- Update GitHub actions to use latest version
- Update Node.js to 20.14.0

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -5,12 +5,10 @@
       "@babel/env",
```
Comment (rupertbates) [inline .babelrc:5]: Do we even need Babel with node 20?

File: .babelrc [hunk 2]
```diff
@@ -5,12 +5,10 @@
...
         }
       }
     ]
   ],
-  "plugins":[
-    "@babel/plugin-proposal-object-rest-spread"
-  ]
-}
\ No newline at end of file
+  "plugins": ["@babel/plugin-proposal-object-rest-spread"]
```
Comment (rupertbates) [inline .babelrc:13]: Don't think we need this:
Screenshot 2024-06-17 at 09 56 15

**Review by rupertbates — APPROVED** [review]
Looks good, I think we can get rid of the rest spread plugin but otherwise 👍 

Would be worth testing all the lambdas on CODE if you haven't already

File: riff-raff.yaml [hunk 3]
```diff
@@ -17,10 +26,4 @@ deployments:
...
         - weekly-fulfilmentUploader-
-       #get from cloudformation
-#  fulfilment-node-stub-cloudformation:
-#  N.B.: cloudformation deployment not quite finished
-#   type: cloud-formation
-#     app: fulfilment-lambdas
-#     parameters:
-#       templatePath: 'cloudformation/cloudformation.yaml'
\ No newline at end of file
+    dependencies: [cloudformation]
```
Comment (paulbrown1982) [inline riff-raff.yaml:29]: Thanks for finishing the job!

File: .babelrc (hunk 2 above)
Comment (AndreaDiotallevi) [reply .babelrc]: Removed thanks.

File: .babelrc (hunk 1 above)
Comment (AndreaDiotallevi) [reply .babelrc:5]: I think we still need `@babel/flow` to transpile flow types.

But yes I agree we should remove it eventually, but it's probably better to do that once we migrate the code to TypeScript & esbuild.

Comment (rupertbates) [reply .babelrc:5]: Yup makes perfect sense

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -29,7 +29,7 @@ jobs:
           zip -r fulfilment-lambdas.zip ./*
           popd
       - name: AWS Auth
-        uses: aws-actions/configure-aws-credentials@v2
+        uses: aws-actions/configure-aws-credentials@v4
```
Comment (johnduffell) [inline .github/workflows/ci.yml:32]: fyi we are supposed to be moving away from this and to v4 of actions-riff-raff
https://github.com/guardian/actions-riff-raff?tab=readme-ov-file#migrating-from-v3-to-v4
might be a good one for next time someone's in tweaking this

**Pull request merged**